lib/features/mailbox/presentation/utils/mailbox_state_manager.dart

```diff
@@ -0,0 +1,44 @@
+
+import 'dart:collection';
+import 'package:core/utils/app_logger.dart';
+import 'package:jmap_dart_client/jmap/core/state.dart' as jmap;
+
+class MailboxStateManager {
+  final Queue<jmap.State> _stateQueue = Queue<jmap.State>();
```

Comment:
- create a `StateQueue` to manage all queue operation: add, remove, ....
- we can reuse it also in `EmailStateManager`
- write tests for `StateQueue`: `add`, `remove` group test. 
  - Cases with multiple states were added to the queue
  - Cases with multiple `removeStatesUpToCurrent`: contain, not contain, queue empty, remove first failed, ...

Reply:
done